litellm/litellm_core_utils/litellm_logging.py

```diff
@@ -1248,6 +1250,20 @@ def success_handler(  # noqa: PLR0915
                                 end_time=end_time,
                                 print_verbose=print_verbose,
                             )
+
+                    if callback == "mlflow":
```

Comment:
this should not be added here. Since the callback is a custom logger, it should work based on line 559

Reply:
make sure the object is instantiated here - https://github.com/BerriAI/litellm/blob/2c37aad1c44af9e98a0af4e36d0b89f2449bf2a9/litellm/litellm_core_utils/litellm_logging.py#L2139

Reply:
Makes sense, thank you for the pointer!

Reply:
@krrishdholakia When I move the logger instantiation to `_init_custom_logger_compatible_class` and `get_custom_logger_compatible_class`, I noticed that the callback is not called for async APIs like `acomplete`. Is there any way to make the callback invoked for async call as well?

https://github.com/BerriAI/litellm/blob/6b03998e324cd063a3d8133c3388ae21572a740d/litellm/litellm_core_utils/litellm_logging.py#L1183-L1198

Reply:
that sounds wrong. 

the custom logger is called across both, can confirm it works as it's used across otel, prometheus, etc. loggers

Reply:
@krrishdholakia But the code indicates the `log_success_event` handler is not invoked for async calls. Am I missing something to make it work?
https://github.com/BerriAI/litellm/blob/6b03998e324cd063a3d8133c3388ae21572a740d/litellm/litellm_core_utils/litellm_logging.py#L1183-L1198

Reply:
@krrishdholakia Figured out - the logger needed to implement `async_log_success_event` and confirmed the logging works after adding it.